lib/salus/scan_report.rb

```diff
@@ -0,0 +1,133 @@
+require 'json'
+
+module Salus
+  class ScanReport
+    attr_reader :scanner_name, :running_time
+
+    def initialize(scanner_name)
+      @scanner_name = scanner_name
+      @passed = nil
+      @running_time = nil
+      @info = {}
+      @errors = []
+    end
+
+    def record
+      started_at = monotime
+
+      yield
+
+      # If the block failed to pass/fail the scan,
+      # default to pass if no errors were recorded
+      @passed = @errors.empty? if @passed.nil?
+    ensure
+      @running_time = (monotime - started_at).round(2)
+    end
+
+    def pass
+      @passed = true
+    end
+
+    def fail
+      @passed = false
+    end
+
+    def passed?
+      @passed == true
+    end
+
+    def failed?
+      @passed == false
+    end
+
+    def info(type, value)
```

Comment:
What if you want to record multiple of the same type of info?
For instance, type might be `dependency` and there could be many values for one project.

Should we append to an array instead and make `@info` `Hash[String=>Array[String]]` ?